### PR TITLE
Add get vault token service for azure

### PR DESCRIPTION
**Pull request by corest**
Towards https://github.com/giantswarm/giantswarm/issues/12862

### CODE REVIEW

File: templates/files/conf/get-vault-token [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+set -uxo pipefail
+
+# Vault configuration
+export VAULT_ADDR=https://{{ .VaultDomainName }}
+export CLUSTER_NAME={{ .ClusterName }}
+
+# Azure metadata
+jwt="$(curl -s 'http://169.254.169.254/metadata/identity/oauth2/token?api-version=2018-02-01&resource=https://management.azure.com' -H Metadata:true | jq -r '.access_token')"
+subscription_id=$(curl -s -H Metadata:true "http://169.254.169.254/metadata/instance?api-version=2017-08-01" | jq -r '.compute | .subscriptionId')
+vm_name=$(curl -s -H Metadata:true "http://169.254.169.254/metadata/instance?api-version=2017-08-01" | jq -r '.compute | .name')
```
Comment (whites11) [inline templates/files/conf/get-vault-token:12]: potentially VM name could have spaces in it, maybe it is better to put double quotes around this variable

```suggestion
vm_name="$(curl -s -H Metadata:true "http://169.254.169.254/metadata/instance?api-version=2017-08-01" | jq -r '.compute | .name')"
```

File: templates/master.yaml.tmpl [hunk 2]
```diff
@@ -813,13 +811,36 @@ systemd:
       WantedBy=multi-user.target
   {{- end }}
 
+  {{ if eq .Provider "azure" -}}
```
Comment (whites11) [inline templates/master.yaml.tmpl:814]: I see you removed the `{{if eq .Provider "aws" }}` elsewhere, why do we still need this unit for azure only?

File: templates/files/conf/get-vault-token (hunk 1 above)
Comment (corest) [reply templates/files/conf/get-vault-token:12]: vm names are master/worker. So we don't have spaces there for sure. But just to be safe - sure

File: templates/master.yaml.tmpl (hunk 2 above)
Comment (corest) [reply templates/master.yaml.tmpl:814]: aws unit us above this one. I just duplicated it to not mess with conditions in the unit body.

Comment (whites11) [reply templates/master.yaml.tmpl:814]: ok

**Review by whites11 — APPROVED** [review]
LGTM

**Pull request merged**